src/tests/unit/cpu/nodes/reorder_node_test.cpp

```diff
@@ -1,266 +1,430 @@
+        reorderNode->executeDynamic(stream);
     }
-    void fillData() override {
-        dstData.resize(getNumElems(dstDims));
-        std::fill(dstData.begin(), dstData.end(), 0);
-        srcData.resize(getNumElems(srcDims));
-        const int numChannels = srcDims[1];
-        const int batchSize = srcDims[2] * srcDims[3] * numChannels;
-        int i = 0;
-        for (int n = 0; n < getNumElems(srcDims); n += batchSize) {
-            for (int c = n; c < n + numChannels; c ++) {
-                for (int sp = c; sp < c + batchSize; sp += numChannels) {
-                    srcData[i++] = static_cast<uint8_t>(sp);
-                }
-            }
-        }
+    void validate(void) {
+        checkReorder(parentEdge->getMemory(), childEdge->getMemory(), prec);
     }
-    std::vector<uint8_t> dstData;
-    std::vector<uint8_t> srcData;
-};
 
-TEST_P(ReorderNSPC2NCSPTest, NSPC2NCSP) {
-    Run();
-}
+    struct BuildReorderParams {
+        ov::intel_cpu::Shape srcShape;
+        ov::intel_cpu::Shape dstShape;
+        LayoutType srcLayout;
+        LayoutType dstLayout;
+    };
 
-TEST_P(ReorderNCSP2NSPCTest, NCSP2NSPC) {
-    Run();
-}
+    void SetUp() override {
+        ReorderCPUTestParamSet reorderTestParam = this->GetParam();
+        BuildReorderParams reorderParams;
+        reorderParams.srcLayout = reorderTestParam.srcLayout;
+        reorderParams.dstLayout = reorderTestParam.dstLayout;
+        reorderParams.srcShape = ov::intel_cpu::Shape(reorderTestParam.inputPartialShape);
+        reorderParams.dstShape = reorderParams.srcShape;
+        inputShapes = reorderTestParam.inputShapes;
+        prec = reorderTestParam.prec;
+
+        buildReorderDynamismGraph(reorderParams);
+    }
 
-const std::vector<bool> forceInplace {false, true};
-const auto NSPC2NCSPparams =::testing::Combine(
-                ::testing::Values(std::vector<size_t> {2, 16, 8, 8}),
-                ::testing::ValuesIn(forceInplace));
+    void buildReorderDynamismGraph(const BuildReorderParams& reorderParams) {
+        BlockedDescCreator::CreatorsMap blockCreatorMap = BlockedDescCreator::getCommonCreators();
+        auto srcBlockedDescCreator = blockCreatorMap[reorderParams.srcLayout];
+        auto dstBlockedDescCreator = blockCreatorMap[reorderParams.dstLayout];
 
-INSTANTIATE_TEST_SUITE_P(smoke_ReorderTestCustomNSPC, ReorderNSPC2NCSPTest, NSPC2NCSPparams,
-                         ReorderCustomImplTestBase::getTestCaseName);
+        inputDesc = srcBlockedDescCreator->createDesc(prec, reorderParams.srcShape);
+        const auto& reorderInputDesc = inputDesc;
+
+        const ov::intel_cpu::CpuBlockedMemoryDesc outputDesc =
+            dstBlockedDescCreator->createDesc(prec, reorderParams.dstShape);
+
+        buildReorderGraph(reorderInputDesc, outputDesc, reorderParams.srcShape, reorderParams.dstShape);
+    }
+
+private:
+    ov::intel_cpu::CpuBlockedMemoryDesc inputDesc{InferenceEngine::Precision::FP32, ov::intel_cpu::Shape{}};
```

Comment:
The store is caused by input node default supported memory descriptor is ncsp, which would cause inputNode->redefineOutputMemory fails because of compatibility.  I have just found cpu plugin inputNode can constructed with external memory descriptor argument, which makes this reorder test graph has compatible selected memory descriptor.
Already updated.